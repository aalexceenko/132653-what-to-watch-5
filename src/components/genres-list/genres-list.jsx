import React from 'react';
import PropTypes from 'prop-types';
import {getGenres} from "../../utils";
import {connect} from "react-redux";
import {ActionCreator} from "../../store/action";

const GenresList = ({activeGenre, genres, changeActiveGenre}) => {

  return (
    <ul className="catalog__genres-list">
      {genres.map((genre, i) => (
        <li className={`catalog__genres-item ${activeGenre === genre ? `catalog__genres-item--active` : ``}`} key={i}>
          <a href="#" className="catalog__genres-link"
            onClick={() => {
              changeActiveGenre(genre);
            }}>{genre}</a>
        </li>
      ))}

    </ul>
  );

};

GenresList.propTypes = {
  changeActiveGenre: PropTypes.func.isRequired,
  activeGenre: PropTypes.string.isRequired,
  genres: PropTypes.array.isRequired,
};

const mapStateToProps = (state) => {
  return ({
    activeGenre: state.activeGenre,
    genres: getGenres(state.films),
  });
};

const mapDispatchToProps = (dispatch) => ({
  changeActiveGenre(newGenre) {
    dispatch(ActionCreator.changeActiveGenre(newGenre));
  }
});

export {GenresList};
export default connect(mapStateToProps, mapDispatchToProps)(GenresList);
