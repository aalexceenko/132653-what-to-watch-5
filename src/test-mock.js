import {getFilmRank} from "./utils";

export const VISIBLE_FILMS_COUNT = 8;
export const ID = 1;
export const VIDEO_CURRENT_TIME = 0;
export const PROGRESS_POSITION = 0;
export const AUTHORIZATION_STATUS = `AUTH`;

export const IS_VIDEO = true;
export const IS_PLAYING = true;
export const IS_IN_MY_LIST = false;
export const TAB = `details`;
export const PREVIEW = `img/aviator.jpg`;
export const PREVIEW_VIDEO = `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;
export const TITLE = `Bohemian Rhapsody`;
export const ACTIVE_GENRE = `Comedy`;
export const GENRES = [`Comedy`, `Crime`, `Documentary`, `Drama`, `Horror`, `Kids & Family`];
export const MATCH = {
  params: {
    id: 1
  }
};

export const FILMS = [
  {
    description: `A young man who was sentenced to seven years`,
    rating: 3.6,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    id: 1,
    title: `Bronson`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    votes: 109661,
    runtime: 92,
    year: 2008,
    ratingText: getFilmRank(3.6),
    myList: false,
    comments: false,
  },
  {
    description: `A young man who was`,
    rating: 3.6,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    id: 2,
    title: `Bronson`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    votes: 109661,
    runtime: 92,
    year: 2008,
    ratingText: getFilmRank(3.6),
    myList: false,
    comments: false,
  },
];

export const FILM = {
  description: `A young man who was sentenced to seven years`,
  rating: 3.6,
  director: `Nicolas Winding Refn`,
  genre: `Action`,
  id: 3,
  title: `Bronson`,
  actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  votes: 109661,
  runtime: 92,
  year: 2008,
  ratingText: getFilmRank(3.6),
  myList: false,
  comments: false,
};

export const REVIEWS = [
  {
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    user: {
      name: `Christina`,
      id: 1,
    },
    comment: `This film is an experience and i has already seen it 4 times`,
  },
  {
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    user: {
      name: `Christina`,
      id: 2,
    },
    comment: `This film is an experience and i has already seen it 4 times`,
  }
];
