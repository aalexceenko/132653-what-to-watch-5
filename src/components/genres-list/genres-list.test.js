import React from "react";
import renderer from "react-test-renderer";
import {GenresList} from "./genres-list";
import {GENRES, ACTIVE_GENRE} from "../../test-mock";

it(`Should GenresList render correctly`, () => {
  const tree = renderer
    .create(
        <GenresList
          genres={GENRES}
          changeActiveGenre={() => {}}
          activeGenre={ACTIVE_GENRE}
        />

    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
