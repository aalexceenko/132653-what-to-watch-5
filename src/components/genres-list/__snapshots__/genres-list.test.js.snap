// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should GenresList render correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Crime
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Documentary
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Horror
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
    >
      Kids & Family
    </a>
  </li>
</ul>
`;
